src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,37 @@
 package core.basesyntax;
 
 public class SalaryInfo {
+    private static final String REPORT_TITLE = "Report for period ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder reportText = new StringBuilder(REPORT_TITLE)
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        User[] usersList = new User[names.length];
+
+        for (int i = 0; i < names.length; i++) {
+            usersList[i] = new User(names[i]);
+        }
+
+        for (String userData : data) {
+            Report userReport = new ReportStringParser().parseReportString(userData);
+            ReportDateUtil reportDate = new ReportDateUtil(userReport.getDate());
+            if (reportDate.checkIsDateInRange(dateFrom, dateTo)) {
+                for (User user : usersList) {
+                    if (user.getUserName().equals(userReport.getUserName())) {
+                        user.addMoneyToSalary(userReport.getSalaryPerHour()
+                                        * userReport.getHoursAmount());
+                        break;
+                    }
+                }
+            }
+        }
+
+        for (User user : usersList) {
+            reportText.append(new UserReportSupplier(user).createUserReportTextLine());
+        }
```

Comment:
iterate over names, and for each name for data, this way you won't need additional arrays and loops